mypy/checker.py

```diff
@@ -2693,12 +2723,31 @@ def flatten(t: Expression) -> List[Expression]:
 
 
 def get_isinstance_type(expr: Expression, type_map: Dict[Expression, Type]) -> Type:
-    type = type_map[expr]
+    all_types = [type_map[e] for e in flatten(expr)]
+
+    types = []  # type: List[Type]
+
+    for type in all_types:
+        if isinstance(type, FunctionLike):
+            if type.is_type_obj():
+                # Type variables may be present -- erase them, which is the best
+                # we can do (outside disallowing them here).
+                type = erase_typevars(type.items()[0].ret_type)
+            types.append(type)
+        elif isinstance(type, TypeType):
+            types.append(type.item)
+        else:  # we didn't see an actual type, but rather a variable whose value is unknown to us
+            return None
 
-    if isinstance(type, TupleType):
-        all_types = type.items
+    assert len(types) != 0
+    if len(types) == 1:
+        return types[0]
     else:
-        all_types = [type]
+        return UnionType(types)
+
+
+def get_issubclass_type(expr: Expression, type_map: Dict[Expression, Type]) -> Type:
```

Comment:
Ah I thought it might be similar, but then I actually managed to make it work without changing anything at all - and didn't notice. I removed it.